feat: align expected lambda headers

adrien-f opened the pull request:
Following latest updates of the AWS Lambda Runtime interfaces, passed headers to the handler are now filtered. `Lambda-Runtime-Cloudevents-Context` is kept for retrocompatibility.

To access the CloudEvent context:

Python

```python
def handler(event, context):
  print(context.client_context.custom)
```

NodeJS/Typsecript

```typescript
exports.handler = async function(event, context) {
  console.log(context.clientContext.Custom)
}
```

Do note that depending on the runtime, you might need to unserialize the custom context first.

See:
https://docs.aws.amazon.com/lambda/latest/dg/runtimes-api.html#runtimes-api-next

adrien-f commented on pkg/converter/cloudevents/cloudevents.go:
```diff
@@ -140,13 +141,22 @@ func (ce *CloudEvent) Request(request []byte, headers http.Header) ([]byte, map[
+		return nil, nil, fmt.Errorf("cannot encode request client context: %w", err)
+	}
+
 	ceContext, err := json.Marshal(context)
```
Any ideas about avoiding a double marshal of the context?

tzununbekov replied:
I have one idea, but it is not for everyone - we can skip marshaling client context and just do this:

```go
	runtimeContext := map[string]string{
		ClientContextKey: fmt.Sprintf("{\"custom\":%s}", ceContext),
		CeContextKey:     string(ceContext),
	}
```

I'll let you decide if this is better than the double marshaling 🙂

adrien-f replied:
Yeah, I thought about something like that but thought I'd keep my dirty tricks for another time :grin: I've pushed this change !

adrien-f commented on pkg/metrics/cloudevents.go:
```diff
@@ -40,7 +40,7 @@ func CETagsFromContext(context map[string]string) (tag.Mutator, tag.Mutator) {
 		return DefaultRequestType, DefaultRequestSource
 	}
-	ceContext, exists := context[cloudevents.ContextKey]
+	ceContext, exists := context[cloudevents.CeContextKey]
```
If one day `CeContextKey` is dropped, more work will need to be done here.

tzununbekov reviewed (approved):
Thank you so much @adrien-f 🙏

adrien-f committed:
feat: align expected lambda headers

Following latest updates of the AWS Lambda Runtime interfaces,
passed headers to the handler are now filtered.

To access the CloudEvent context:

Python

```python
def handler(event, context):
  print(context.client_context.custom)
```

NodeJS/Typsecript

```typescript
exports.handler = async function(event, context) {
  console.log(context.clientContext.Custom)
}
```

Do note that depending on the runtime, you might need to unserialize
the custom context first.

See:
https://docs.aws.amazon.com/lambda/latest/dg/runtimes-api.html#runtimes-api-next
Signed-off-by: Adrien Fillon <adrien.fillon@manomano.com>
